api.planx.uk/modules/flows/validate/service/templatedNodes.test.ts

```diff
@@ -0,0 +1,159 @@
+import { ComponentType } from "@opensystemslab/planx-core/types";
+import { queryMock } from "../../../../tests/graphqlQueryMock.js";
+import { validateTemplatedNodes } from "./templatedNodes.js";
+import { userContext } from "../../../auth/middleware.js";
+import { getTestJWT } from "../../../../tests/mockJWT.js";
+
+beforeAll(() => {
+  const getStoreMock = vi.spyOn(userContext, "getStore");
+  getStoreMock.mockReturnValue({
+    user: {
+      sub: "123",
+      jwt: getTestJWT({ role: "teamEditor" }),
+    },
+  });
+});
+
+test("returns status 'Not applicable' if this is not a templated flow", async () => {
+  queryMock.mockQuery({
+    name: "GetTemplatedFlowEdits",
+    matchOnVariables: false,
+    data: {
+      flow: {
+        templatedFrom: null,
+        edits: null,
+      },
+    },
+  });
+
+  const result = await validateTemplatedNodes(
+    "templated-flow-123",
+    mockTemplatedFlow,
+  );
+  expect(result).toEqual({
+    title: "Templated nodes",
+    status: "Not applicable",
+    message: "This is not a templated flow",
+  });
+});
+
+test("returns status 'Fail' if no required nodes have been edited", async () => {
+  queryMock.mockQuery({
+    name: "GetTemplatedFlowEdits",
+    matchOnVariables: false,
+    data: {
+      flow: {
+        templatedFrom: "source-template-123",
+        edits: { data: {} },
+      },
+    },
+  });
+
+  const result = await validateTemplatedNodes(
+    "templated-flow-123",
+    mockTemplatedFlow,
+  );
+  expect(result).toEqual({
+    title: "Templated nodes",
+    status: "Fail",
+    message: `Customise each "Required" node before publishing your templated flow`,
+  });
+});
+
+test("returns status 'Pass' if the required node has been directly edited", async () => {
+  queryMock.mockQuery({
+    name: "GetTemplatedFlowEdits",
+    matchOnVariables: false,
+    data: {
+      flow: {
+        templatedFrom: "source-template-123",
+        edits: { data: mockEditParent },
+      },
+    },
+  });
+
+  const result = await validateTemplatedNodes(
+    "templated-flow-123",
+    mockTemplatedFlow,
+  );
+  expect(result).toEqual({
+    title: "Templated nodes",
+    status: "Pass",
+    message: `All "Required" nodes in your templated flow have been customised`,
+  });
+});
+
+test("returns status 'Pass' if the child of a required node has been edited", async () => {
+  queryMock.mockQuery({
+    name: "GetTemplatedFlowEdits",
+    matchOnVariables: false,
+    data: {
+      flow: {
+        templatedFrom: "source-template-123",
+        edits: { data: mockEditOption },
+      },
+    },
+  });
+
+  const result = await validateTemplatedNodes(
+    "templated-flow-123",
+    mockTemplatedFlow,
+  );
+  expect(result).toEqual({
+    title: "Templated nodes",
+    status: "Pass",
+    message: `All "Required" nodes in your templated flow have been customised`,
+  });
+});
+
+const mockTemplatedFlow = {
+  _root: {
+    edges: ["Question"],
+  },
+  Question: {
```

Comment:
This node mock comes directly from the LDC source template that was failing :heavy_check_mark: